addons/point_of_sale/static/src/app/screens/product_screen/product_screen.js

```diff
@@ -405,9 +405,25 @@ export class ProductScreen extends Component {
             ? this.getProductsByCategory(this.pos.selectedCategory)
             : this.products;
 
-        return products.filter((p) =>
-            unaccent(p.searchString, false).toLowerCase().includes(words)
+        const filteredProducts = products.filter((p) =>
+            unaccent(p.searchString, false)
+                .toLowerCase()
+                .includes(unaccent(words, false).toLowerCase())
         );
+
+        return filteredProducts.sort((a, b) => {
+            const nameA = unaccent(a.searchString, false).toLowerCase();
+            const nameB = unaccent(b.searchString, false).toLowerCase();
+            const searchWord = unaccent(words, false).toLowerCase();
+
+            const indexA = nameA.indexOf(searchWord);
+            const indexB = nameB.indexOf(searchWord);
+
+            // Sort by match index, push non-matching items to the end, and use alphabetical order as a tiebreaker
+            return (
+                (indexA === -1) - (indexB === -1) || indexA - indexB || nameA.localeCompare(nameB)
```

Comment:
I don't really see how `indexA` or `indexB` can be `=== -1` as they already passed the first filter so I'd say the first comparaison is not really useful what do you think ?
```suggestion
                indexA - indexB || nameA.localeCompare(nameB)
```